MetaMask/metamask-extension
# feat: Add Confirm ScrollToBottom component and update ConfirmPage

## **Description**

Fixes: https://github.com/MetaMask/metamask-extension/issues/21285

Creates a ScrollToBottom component for the ConfirmPage which reuses the ScrollToBottom hook

Features:
- button only appears when the content is scrollable
- button click scrolls to the bottom
- button disappears when the user reaches the bottom
- sets confirm.isScrollToBottomNeeded to be used to disable/enable footer confirm button

ScrollToBottom hook Updates:
- add hasScrolledToBottom state while keeping isScrolledToBottom state allowing us to integrate the ability for the button to reappear after it disappears
- add smooth scrolling

Additional Updates: 
- new `isScrollToBottomNeeded` prop on Confirmation type
- adds isScrollToBottomNeeded to confirm state

For confirmation page only, overrides ButtonIcon hover styles as hover styles have an outstanding issue with our Design System components. Currently, the hover style incorporates the opacity with the color which causes issues. (ref: https://github.com/MetaMask/metamask-extension/pull/19976/files)

> [!NOTE]  
> no behavioral changes have been made to the other instances that use the ScrollToBottom hook except they will not use smooth scroll rather than instant scroll 

---
Dev note:
 
Regarding the hover style, for the former design with only the opacity change.

On ButtonIcon hover, the background color is changing with the intended opacity change
```
    --color-background-hover: #00000005;
    --color-background-pressed: #00000014;
```
ButtonIcon PR: [https://github.com/MetaMask/metamask-extension/pull/19976/files](https://github.com/MetaMask/metamask-extension/pull/19976/files)

The background color is set in these design system tokens, which caused an issue when we only wanted to target opacity. 

## **Manual testing steps**


1. open storybook `yarn storybook`
2. see http://localhost:6006/?path=/docs/components-app-confirm-scrolltobottom--docs

## **Screenshots/Recordings**

![Screenshot 2024-02-07 at 3 49 44 PM](https://github.com/MetaMask/metamask-extension/assets/20778143/61a3dd81-b0e3-467f-8936-5806d1812582)
![confirm-scrolltobottom](https://github.com/MetaMask/metamask-extension/assets/20778143/e1b8e4eb-f5e8-4756-a6f8-5cd1824d6210)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
